x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java

```diff
@@ -155,6 +177,89 @@ static EqlSearchResponse createResponse(Results results, AsyncExecutionId id) {
+    }
+
+    private static List<EqlSearchResponse.Event> qualifyEvents(List<EqlSearchResponse.Event> events, String clusterAlias,
+                                                               boolean emptyListIfNoEvents) {
+        List<EqlSearchResponse.Event> qualifiedEvents = null;
+        if (events != null && events.isEmpty() == false) {
+            qualifiedEvents = new ArrayList<>(events.size());
+            for (EqlSearchResponse.Event e : events) {
+                qualifiedEvents.add(new EqlSearchResponse.Event(buildRemoteIndexName(clusterAlias, e.index()), e.id(), e.source(),
+                    e.fetchFields()));
+            }
+        }
+        return qualifiedEvents == null && emptyListIfNoEvents ? new ArrayList<>() : qualifiedEvents;
+    }
+
+    private static Exception qualifyException(Exception e, String[] indices, String clusterAlias) {
+        Exception finalException = e;
+        if (e instanceof RemoteTransportException && e.getCause() instanceof IndexNotFoundException) {
+            IndexNotFoundException infe = (IndexNotFoundException) e.getCause();
+            if (infe.getIndex() != null) {
+                String qualifiedIndex;
+                String exceptionIndexName = infe.getIndex().getName();
+                final String UNKNOWN_INDEX_PREFIX = "Unknown index [";
+                if (exceptionIndexName.startsWith(UNKNOWN_INDEX_PREFIX)) {
+                    int offsetOfClosingBraket = exceptionIndexName.indexOf(']', UNKNOWN_INDEX_PREFIX.length());
+                    String infeIndices = exceptionIndexName.substring(UNKNOWN_INDEX_PREFIX.length(), offsetOfClosingBraket);
+                    String[] notFoundIndices;
+                    // see RestEqlSearchAction#prepareRequest() or GH#63529 for an explanation of "*,-*" replacement
+                    notFoundIndices = infeIndices.equals("*,-*") ? indices : infeIndices.split(",");
+                    StringJoiner sj = new StringJoiner(",");
+                    for (int i = 0; i < notFoundIndices.length; i ++) {
+                        sj.add(buildRemoteIndexName(clusterAlias, notFoundIndices[i]));
+                    }
+                    qualifiedIndex = sj.toString();
+                } else {
+                    qualifiedIndex = buildRemoteIndexName(clusterAlias, exceptionIndexName);
+                }
+                // This will expose a "uniform" failure root_cause, with same "type" ("index_not_found_exception") and "reason" ("no such
+                // index [...]"); this is also similar to a non-CCS `POST inexistent/_eql/search?ignore_unavailable=false`, but
+                // unfortunately unlike an inexistent pattern search: `POST inexistent*/_eql/search?ignore_unavailable=false, which raises a
+                // VerificationException as it's root cause. I.e. the failures are not homogenous.
+                finalException = new RemoteTransportException(e.getMessage(), new IndexNotFoundException(qualifiedIndex));
```

Comment:
When @astefan come back please raise this with him since he worked on something similar. It's worth creating an issue to see whether the two errors can be aligned and if not why?

Reply:
Great. It would seem good to me to return/report a not found index in the same way; i.e. no matter if the verifier finds the query error or other layer under EQL. (This could probably also be a subsequent PR.)